h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,261 @@
+package hex.genmodel.tools;
+
+import com.mxgraph.layout.mxCompactTreeLayout;
+import com.mxgraph.layout.mxEdgeLabelLayout;
+import com.mxgraph.layout.mxIGraphLayout;
+import com.mxgraph.util.mxCellRenderer;
+import hex.genmodel.algos.tree.ConvertTreeOptions;
+import hex.genmodel.algos.tree.SharedTreeGraph;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.genmodel.algos.tree.TreeBackedMojoModel;
+import org.jgrapht.ext.JGraphXAdapter;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedMultigraph;
+import org.jgrapht.io.*;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    public static void main(String[] args) {
+        MojoPrinter mojoPrinter;
+        if ("png".equals(getFormat(args))){
+            mojoPrinter = new JgraphtPrintMojo();
+        } else {
+            mojoPrinter = new PrintMojo();
+        }
+        mojoPrinter.parseArgs(args);
+
+        // Run the main program
+        try {
+            mojoPrinter.run();
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.exit(2);
+        }
+        
+        // Success
+        System.exit(0);
+    }
+    
+    static String getFormat(String[] args) {
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("--format")) {
+                return args[++i];
+            }
+        }
+        return "";
+    }
+    
+    protected static void usage() {
```

Comment:
is this copy-paste really necessary? it will be a pain to maintain

Reply:
not necessary with SPI approach